lib/ansible/modules/service.py

```diff
@@ -16,6 +16,14 @@
 description:
     - Controls services on remote hosts. Supported init systems include BSD init,
       OpenRC, SysV, Solaris SMF, systemd, upstart.
+    - This module acts as a proxy to the underlying service manager module. While all arguments will be passed to the
+      underlying module, not all modules support the same arguments. This documentation only covers the minimum intersection
+      of module arguments that all service manager modules support.
+    - This modules manages services on a target without specifying a service manager module
+      (like M(ansible.builtin.systemd), M(ansible.builtin.sysvinit), and so on).
+      It is convenient to use in an heterogeneous environment of machines without having to create a specific task for
+      each service manager. M(ansible.builtin.service) calls behind the module for the service manager used by the operating system
+      discovered by the module M(ansible.builtin.setup).  If `setup` was not yet run, `service` will run it.
```

Comment:
```suggestion
    - This module is a proxy for multiple more specific service manager modules
      (such as M(ansible.builtin.systemd) and M(ansible.builtin.sysvinit)).
      This allows management of a heterogeneous environment of machines without creating a specific task for
      each service manager. The module to be executed is determined by the I(use) option, which defaults to the
      service manager discovered by M(ansible.builtin.setup).  If `setup` was not yet run, this module may run it.
```

The default module and whether or not the setup module is run depend on the value of `use`, so it might be better to improve the option documentation and remove some of the detail here.